modules/post/osx/gather/password_prompt_spoof.rb

```diff
@@ -106,6 +106,19 @@ def run
       print_good("password file contents: #{password_data}")
       passf = store_loot("password", "text/plain", session, password_data, "passwd.pwd", "OSX Password")
       print_good("Password data stored as loot in: #{passf}")
+      pwd = password_data.split(':', 3)
```

Comment:
why not just say 'pwd = password_data.split(':')[2]' ? It looks like this method also stores an array as the password, not the string, since it doesn't call `.shift()` again: `["password"]`

Reply:
I wondered the same thing, but considered that perhaps the password could contain `:` which might cause issues.

```
irb(main):001:0> password_data = "some date:some username:my password which:has some colons:because why not"
=> "some date:some username:my password which:has some colons:because why not"
irb(main):002:0> pwd = password_data.split(':', 3)
=> ["some date", "some username", "my password which:has some colons:because why not"]
irb(main):003:0> pwd.shift() # date
=> "some date"
irb(main):004:0> pwd.shift() # username
=> "some username"
irb(main):005:0> pwd
=> ["my password which:has some colons:because why not"]
irb(main):006:0> # versus:
=> nil
irb(main):007:0> password_data.split(':')[2]
=> "my password which"
irb(main):008:0> 
```

I say that, without having looked at what the existing data format looks like, and whether this scenario is even possible.

Reply:
What I actually discovered was the data is `<date>:<un>:<pass>` and originally wrote the split as suggested. However if the password was `hi:` I couldn't get it reliably due to the split lopping off the last one. Worse, `hi:::` was just a nightmare. The code I used will keep all trailing colons without having to reconstruct later

Reply:
In the below case, the use entered the password "a:b:::", yes their password had trailing colons.
```
irb(main):002:0> "20190416_195206:victim:a:b:::".split(':', 3)
=> ["20190416_195206", "victim", "a:b:::"]
irb(main):003:0> "20190416_195206:victim:a:b:::".split(':')
=> ["20190416_195206", "victim", "a", "b"]
```
`<date>:<un>:<pass>`
This is something I need to test in the jtr modules as well now.  Since the person used a password which is the same as our delimiter, a regular `.split(':')[2].join(':')` or some combination of taking the 2+th elements and joining with a ':' would still miss all trailing colons.  Since the format is driven by us, its not likely to change, so this was the safest way I could figure to do it.

Reply:
I also determined that sending an array with just a string in it was accepted by the `create_credential` library, so I saved a few characters of code by not adding the last shift (but can change it if you want)

Reply:
Alright, thanks for the clarification.